package.json

```
@@ -4,19 +4,23 @@
   "license": "MIT",
   "scripts": {
     "angular:build-generator": "cd node_modules/devextreme-angular-generator && npm i --no-audit --no-fund && npm run build:local",
-    "angular:discover-metadata": "npm run tools:discover-declarations && npm run tools:generate-imd -- --output-path ./packages/devextreme-angular/metadata/integration-data.json",
-    "angular:generate": "npm run angular:discover-metadata && npm run angular:build-generator && cd packages/devextreme-angular && npm run generate",
+    "angular:copy-metadata": "mkdir -p ./packages/devextreme-angular/metadata && cp ./artifacts/internal-tools/NGMetaData.json ./packages/devextreme-angular/metadata/",
+    "angular:generate": "nx regenerate devextreme-angular",
     "react:build-generator": "cd node_modules/devextreme-react-generator && npm i --no-audit --no-fund && npm run build:local",
-    "react:discover-metadata": "npm run tools:discover-declarations && npm run tools:generate-imd -- --output-path ./packages/devextreme-react/metadata/integration-data.json",
-    "react:generate": "npm run react:discover-metadata && npm run react:build-generator && cd packages/devextreme-react && npm run generate",
+    "react:copy-metadata": "mkdir -p ./packages/devextreme-react/metadata && cp ./artifacts/internal-tools/integration-data.json ./packages/devextreme-react/metadata/",
+    "react:generate": "nx regenerate devextreme-react",
     "tools:discover-declarations": "dx-tools discover --js-scripts ./packages/devextreme/js --exclude \"js/(renovation|__internal|.eslintrc.js)\" --compiler-options \"{ \\\"typeRoots\\\": [] }\" --artifacts ./artifacts/internal-tools",
     "tools:generate-imd": "dx-tools integration-data-generator --artifacts ./artifacts/internal-tools",
+    "tools:generate-ng-smd": "dx-tools generate-ng-smd --artifacts ./artifacts/internal-tools",
+    "tools:discover-metadata": "npm run tools:discover-declarations && npm run tools:generate-imd && npm run tools:generate-ng-smd",
     "vue:build-generator": "cd node_modules/devextreme-vue-generator && npm i --no-audit --no-fund && npm run build:local",
-    "vue:discover-metadata": "npm run tools:discover-declarations && npm run tools:generate-imd -- --output-path ./packages/devextreme-vue/metadata/integration-data.json",
-    "vue:generate": "npm run vue:discover-metadata && npm run vue:build-generator && cd packages/devextreme-vue && npm run generate",
+    "vue:copy-metadata": "mkdir -p ./packages/devextreme-vue/metadata && cp ./artifacts/internal-tools/integration-data.json ./packages/devextreme-vue/metadata/",
+    "vue:generate": "nx regenerate devextreme-vue",
+    "generate": "nx run-many -t regenerate",
```

```suggestion
    "regenerate": "nx run-many -t regenerate",
```

Maybe use `regenerate` in all cases?